charger/daheimladen-mb.go

```diff
@@ -167,9 +182,20 @@ func (wb *DaheimLadenMB) Status() (api.ChargeStatus, error) {
 	if err != nil {
 		return api.StatusNone, err
 	}
-
 	s := binary.BigEndian.Uint16(b)
-
+	// connector state lesen und auf Änderung prüfen (0 = nicht verbunden, 1 = verbunden)
+	if cb, err := wb.conn.ReadHoldingRegisters(dlRegConnectorState, 1); err == nil {
+		connState := binary.BigEndian.Uint16(cb)
+		if connState != wb.connState {
+			// Connector hat gewechselt -> wenn Wallbox Phasenumschaltung unterstützt, dann phases=3
+			if wb.hasPhaseSwap {
+				wb.phases = 3
+			} 
+            wb.connState = connState
+		}
+	} else {
+		wb.log.TRACE.Println("read connector state:", err)
+	}
 	switch s {
 	case 1: // Standby (A)
 		return api.StatusA, nil
```

Comment:
```suggestion
		wb.phases = 3
		return api.StatusA, nil
```

Hätte exakt den gleichen Effekt wie alle deine anderen Änderungen zusammen ;)

Leider bringt das aber imho alles nichts, da evcc derzeit nicht damit rechnet dass der Charger eigenmächtig die Phasenkonfiguration geändert hat bzw. immer auf 3 nach abstecken zurückändert.

Überlegung: Wenn wir getPhases() (sofern implementiert) gezielt vor Enable() abfragen würden dann wäre das behebbar @andig

Reply:
Solches Eigenleben ist immer MIST

Reply:
>Hätte exakt den gleichen Effekt wie alle deine anderen Änderungen zusammen ;)

Nicht unbedingt, oder? Die Implementierung wird ja auch für die alten Daheimlader (ohne Phasenumschaltung) genutzt. Was ist, wenn die mittels Lasttrennschalter auf 1p stehen?

Reply:
> bzw. immer auf 3 nach abstecken zurückändert.

nicht nur nach abstecken, auch nach anstecken.

Reply:
Das Eine hat mit dem Anderen nichts zu tun.
Die Phasenumschaltung ist in der Implementierung schon nur optional.
Bei Alt-Boxen werden die API-Schnittstellen gar nicht dekoriert.

Um DIESES eigenmächtige Verhalten hier zu fixen müsste imho der LP angepasst werden. Das geht im Charger selber nicht.

Reply:
>Um DIESES eigenmächtige Verhalten hier zu fixen müsste imho der LP angepasst werden. Das geht im Charger selber nicht.

Also eine Ausnahme im Loadpoint-Code nur für Daheimladen? Das erscheint mir unrealistisch.

Reply:
oder wäre das "pauschal" umsetzbar, das bei vehicleConnect der Phasenzustand des 1p3pCharger abgefragt wird?

Reply:
Ja, ich weiß, ist aber schwierig.

Ich mach hier zu, in der Hoffnung, dass #23749 die Lösung bringt.
Vielen Dank.